android/app/src/main/java/app/priceguard/ui/home/ProductSummaryAdapter.kt

```diff
@@ -0,0 +1,79 @@
+package app.priceguard.ui.home
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import app.priceguard.R
+import app.priceguard.databinding.ItemProductSummaryBinding
+
+class ProductSummaryAdapter :
+    ListAdapter<ProductSummary, ProductSummaryAdapter.ViewHolder>(diffUtil) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val binding =
+            ItemProductSummaryBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return ViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val item = currentList[position]
+        holder.bind(item)
+    }
+
+    class ViewHolder(private val binding: ItemProductSummaryBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(item: ProductSummary) {
+            binding.setBinding(item)
+        }
+
+        private fun ItemProductSummaryBinding.setBinding(item: ProductSummary) {
+            summary = item
+            when (item) {
+                is ProductSummary.RecommendedProductSummary -> {
+                    tvProductRecommendRank.visibility = View.VISIBLE
+                    msProduct.visibility = View.GONE
+                    setRecommendRank(item)
+                }
+
+                is ProductSummary.UserProductSummary -> {
+                    tvProductRecommendRank.visibility = View.GONE
+                    msProduct.visibility = View.VISIBLE
+                    setSwitchListener()
+                }
+            }
+        }
+
+        private fun ItemProductSummaryBinding.setSwitchListener() {
+            msProduct.setThumbIconResource(if (msProduct.isChecked) R.drawable.ic_notifications_active else R.drawable.ic_notifications_off)
```

Comment:
어짜피 default가 on 이면 off 일때만 resource를 변경해도 되지 않을까요?
```suggestion
            if (!msProduct.isChecked) {
                msProduct.setThumbIconResource(R.drawable.ic_notifications_off)
            }
```